cmd/create/service/cmd.go

```diff
@@ -0,0 +1,360 @@
+  Use this command to create managed services.`,
+	Example: `  # Create a Managed Service of type service1.
+  rosa create service --type=service1 --name=clusterName`,
+	Run:                run,
+	Hidden:             true,
+	DisableFlagParsing: true,
+	Args: func(cmd *cobra.Command, argv []string) error {
+		err := arguments.ParseUnknownFlags(cmd, argv)
+		if err != nil {
+			return err
+		}
+
+		if len(cmd.Flags().Args()) > 0 {
+			return fmt.Errorf("Unrecognized command line parameter")
+		}
+		return nil
+	},
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.SortFlags = false
+
+	// Basic options
+	flags.StringVar(
+		&args.ServiceType,
+		"type",
+		"",
+		"Type of service.",
+	)
+
+	flags.StringVar(
+		&args.ClusterName,
+		"name",
+		"",
+		"Name of the service instance.",
+	)
+
+	arguments.AddRegionFlag(flags)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Parse out CLI flags, then override positional arguments
+	// This allows for arbitrary flags used for addon parameters
+	_ = cmd.Flags().Parse(argv)
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	awsClient := aws.GetAWSClientForUserRegion(reporter, logger)
+
+	// Get AWS region
+	args.AwsRegion, err = aws.GetRegion(arguments.GetRegion())
+	if err != nil {
+		reporter.Errorf("Error getting region: %v", err)
+		os.Exit(1)
+	}
+	reporter.Debugf("Using AWS region: %q", args.AwsRegion)
+
+	awsCreator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Unable to get IAM credentials: %v", err)
+		os.Exit(1)
+	}
+	reporter.Debugf("Using AWS creator: %q", awsCreator.ARN)
+
+	args.AwsAccountID = awsCreator.AccountID
+	args.Properties = map[string]string{
+		properties.CreatorARN: awsCreator.ARN,
+		properties.CLIVersion: info.Version,
+	}
+
+	// Openshift version to use.
+	versionList, err := getVersionList(ocmClient)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+	version := versionList[0]
+	minor := ocm.GetVersionMinor(version)
+
+	// Add-on parameter logic
+	addOn, err := ocmClient.GetAddOn(args.ServiceType)
+	if err != nil {
+		reporter.Errorf("Failed to get add-on %q: %s", args.ServiceType, err)
+		os.Exit(1)
+	}
+	parameters := addOn.Parameters()
+
+	if parameters.Len() > 0 {
+		args.Parameters = map[string]string{}
+		// Determine if all required parameters have already been set as flags.
+		if arguments.HasUnknownFlags() {
+			parameters.Each(func(param *cmv1.AddOnParameter) bool {
+				flag := cmd.Flags().Lookup(param.ID())
+				if param.Required() && (flag == nil || flag.Value.String() == "") {
```

Comment:
I think the issue here is the `if arguments.HasUnknownFlags()` - since these are addon parameters, and we don't support interactive flow, I think we need to remove that conditional check and run the loop for _all_ parameters defined in the addon.